zebra-network/src/peer/handshake.rs

```diff
@@ -372,11 +372,10 @@ where
     ///
     /// If this is unset, the minimum accepted protocol version for peer connections is kept
     /// constant over network upgrade activations.
-    pub fn with_best_tip_height(
-        mut self,
-        best_tip_height: Option<watch::Receiver<Option<block::Height>>>,
-    ) -> Self {
-        self.best_tip_height = best_tip_height;
+    ///
+    /// Use [`NoChainTip`] to explicitly provide no chain tip.
+    pub fn with_chain_tip_receiver(mut self, chain_tip_receiver: C) -> Self {
+        self.chain_tip_receiver = Some(chain_tip_receiver);
         self
     }
```

Comment:
Optional idea: The `chain_tip_receiver` field can be changed to not be an `Option` in the `Builder` as well. But this setter method would have to change the generic type parameters. Unfortunately, Rust doesn't help much here because the `struct` update pattern (`..rest`) isn't available yet if the generic types differ ([RFC](https://github.com/rust-lang/rfcs/blob/master/text/2528-type-changing-struct-update-syntax.md), [tracking issue](https://github.com/rust-lang/rust/issues/86555)).

```suggestion
    ///
    /// Use [`NoChainTip`] to explicitly provide no chain tip.
    pub fn with_chain_tip_receiver<NewC>(self, chain_tip_receiver: NewC) -> Builder<S, NewC>
    where
        NewC: ChainTip,
    {
        Builder {
            chain_tip_receiver,
            // TODO: Until Rust RFC 2528 reaches stable, we can't do `..self`
            config: self.config,
            inbound_service: self.inbound_service,
            timestamp_collector: self.timestamp_collector,
            our_services: self.out_services,
            user_agent: self.user_agent,
            relay: self.relay,
            inv_collector: self.inv_collector,
        }
    }
```

Reply:
63d87349